src/core/ddsc/tests/psmxif.c

```diff
@@ -0,0 +1,441 @@
+    ddsrt_free(configstr);
+    configstr_in = xconfigstr;
+  }
+  const dds_entity_t domain = dds_create_domain (domainId, configstr_in);
+  CU_ASSERT_FATAL(domain <= 0);
+}
+
+/// @brief Check that shared memory can be enabled (when supported) and the used locator can be set.
+/// @methodology
+/// - Check that the data types I'm planning to use are actually suitable for use with shared memory.
+/// - Expectation: They are memcopy-safe.
+/// 
+/// - Create a configuration with a psmx interface and specify the locator.
+/// - Create a domain using this configuration.
+/// - Check the locator used by the psmx instance.
+/// - Expectation: The locator is the same as specified in the config for the domain.
+/// - Query whether shared memory is supported.
+/// - Assert that there is exactly one psmx instance.
+/// - Assert that the psmx instance has a nonempty instance_name.
+/// - Create some entities
+/// - Check if shared memory is enabled.
+/// - Expectation: Shared memory is enabled iff the psmx interface supports it (use queried value).
+/// - Delete the domain
+/// - Check the function call counts of the dummy psmx.
+/// - Expectation: The counts match expectations. In particular, create counts must match their delete counterpart.
+/// 
+/// - Create a configuration with a psmx interface capable of shared memory and don't specify a locator.
+/// - Create a domain using this configuration.
+/// - Query whether shared memory is supported.
+/// - Assert that there is exactly one psmx instance.
+/// - Assert that the psmx instance has a nonempty instance_name.
+/// - Create some entities
+/// - Check if shared memory is enabled.
+/// - Expectation: Shared memory is enabled iff the psmx interface supports it (use queried value).
+/// - Delete the domain
+/// - Check the function call counts of the dummy psmx.
+/// - Expectation: The counts match expectations. In particular, create counts must match their delete counterpart.
+/// 
+CU_Test(ddsc_psmxif, shared_memory)
+{
+  dummy_mockstats_t* dmock = NULL;
+  char strbuf[512];
+  const size_t strbuf_size = sizeof(strbuf);
+  {
+    // Check that the data types I'm planning to use are actually suitable for use with shared memory.
+    dds_data_type_properties_t props;
+    props = dds_stream_data_types(SC_Model_desc.m_ops);
+    CU_ASSERT_FATAL((props & DDS_DATA_TYPE_IS_MEMCPY_SAFE) == DDS_DATA_TYPE_IS_MEMCPY_SAFE);
+    props = dds_stream_data_types(PsmxType1_desc.m_ops);
+    CU_ASSERT_FATAL((props & DDS_DATA_TYPE_IS_MEMCPY_SAFE) == DDS_DATA_TYPE_IS_MEMCPY_SAFE);
+  }
+
+  const dds_domainid_t domainId = 0;
+  bool supports_shared_memory_expected = true;
+  dds_psmx_endpoint_t* psmx_endpt_expected = NULL;
+  dds_psmx_topic_t* psmx_topic_expected = NULL;
+  bool specify_locator[] = {true, false};
+  uint8_t locator_in[16];
+  memset(locator_in, 0x0, sizeof(locator_in)); // Avoid warning 'uninitialized value'.
+  ((uint64_t*)locator_in)[0] = (uint64_t)0x4a4d203df6996395;
```

Comment:
These are not meaningful anymore and so should go — but I also want to point out that `uint8_t a[16];*((uint64_t*)a)=...` risks misaligned access, and with that, undefined behaviour. In this case, rather than a `memset` followed by two 64-bit integer assignments, you could just have written out the value as an initializer, or.

Reply:
The locator is now part of the config string.